pkg/system/phase2_creating.go

```diff
@@ -171,6 +176,128 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	return nil
 }
 
+func (r *Reconciler) UpgradeDbPostgres16() error {
+	var err error = nil
+
+	if r.NooBaa.Status.PostgresUpdatePhase == nbv1.UpgradePhaseFinished {
+		r.Logger.Infof("UpgradePostgresDB: DB is already upgraded to postgresql-16")
+		return nil
+	}
+
+	switch r.NooBaa.Status.PostgresUpdatePhase {
+	case "":
+		// Check whether it is fresh install case
+		sts := util.KubeObject(bundle.File_deploy_internal_statefulset_postgres_db_yaml).(*appsv1.StatefulSet)
+		sts.Name = "noobaa-db-pg"
+		sts.Namespace = r.NooBaa.Namespace
+		stsObj, err := util.KubeGetObject(sts)
+		// Postgres database doesn't exists, no need to upgrade.
+		if err != nil {
```

Comment:
I think that this check should be outside of the `switch`

Reply:
Empty phrase is used for the cases when we move from older version. On older version `upgradephase` flag was not used/updated

Reply:
Moved this part out of switch